2017-01-26  Richard Biener  <rguenther@suse.de>

	Backport from mainline
	2016-09-03  Kirill Yukhin  <kirill.yukhin@intel.com>

	* ubsan.c (ubsan_use_new_style_p): Fix check for empty string.


git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/branches/gcc-6-branch@244923 138bc75d-0d04-0410-961f-82ee72b054a4

diff --git a/gcc/ubsan.c b/gcc/ubsan.c
index 56637d8..1093824 100644
--- a/gcc/ubsan.c
+++ b/gcc/ubsan.c

@@ -1471,7 +1471,7 @@
 
   expanded_location xloc = expand_location (loc);
   if (xloc.file == NULL || strncmp (xloc.file, "\1", 2) == 0
-      || xloc.file == '\0' || xloc.file[0] == '\xff'
+      || xloc.file[0] == '\0' || xloc.file[0] == '\xff'
       || xloc.file[1] == '\xff')
     return false;
 

